concepts/ownership/about.md

```diff
@@ -1,118 +1,146 @@
 }
+```
+when A goes out of scope, its dictionary will be automatically squashed, and the program will compile.
+without the `Destruct` trait, the code won't compile. because dictionaries won't be able to squashed.
 
+## Return Values and Scope
+Returning values is equivalent to moving them. the example below shows an example of a function that returns some value.
 
-```
-<figure> Using snapshots in Cairo allows you to maintain ownership of variables while safely accessing their state at specific points in time, ensuring data integrity and preventing accidental modifications.</figure>
+``` rust
+#[derive(Drop)]
+struct A {}
+
+fn main() {
+    let a1 = gives_ownership();           // gives_ownership moves its return
+                                          // value into a1
 
-### Desnap Operator:
+    let a2 = A {};                        // a2 comes into scope
 
- - Convert a snapshot back to a regular variable with `*`.
+    let a3 = takes_and_gives_back(a2);    // a2 is moved into
+                                          // takes_and_gives_back, which also
+                                          // moves its return value into a3
 
+} // Here, a3 goes out of scope and is dropped. a2 was moved, so nothing
+  // happens. a1 goes out of scope and is dropped.
 
+fn gives_ownership() -> A {               // gives_ownership will move its
+                                          // return value into the function
+                                          // that calls it
 
-### Mutable References
-Mutable references allow you to pass mutable values to a function and have them implicitly returned, maintaining ownership in the calling context. This can be achieved using the `ref` modifier, provided the variable is declared as mutable with `mut`.
+    let some_a = A {};                    // some_a comes into scope
 
-```rust
-fn get_len(arr: @Array<u128>) -> u32{ //changing the parameter to take in a snapshot rather than the ownership
-    arr.len()
+    some_a                                // some_a is returned and
+                                          // moves ownership to the calling
+                                          // function
 }
-fn foo(ref arr: Array<u128>) { //use the ref modifier in accepting argument
 
-    arr.pop_front();
+// This function takes an instance some_a of A and returns it
+fn takes_and_gives_back(some_a: A) -> A { // some_a comes into scope
+
+    some_a                                // some_a is returned and 
+                                          // moves ownership to the calling
+                                          // function
 }
+```
+While this works, moving into and out of every function is a bit tedious. What if we want to let a function use a value but not move the value? It’s quite annoying that anything we pass in also needs to be passed back if we want to use it again, in addition to any data resulting from the body of the function that we might want to return as well.
```

Comment:
Let's keep this focused just on _moving_ return values, remove everything after the above example to the end of the file